File: backend/src/main/java/isep/webapp/terasync/model/odSystem/ProductMenuList.java
package isep.webapp.terasync.model.odSystem;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class ProductMenuList extends isep.webapp.terasync.model.Entity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productMenuListId")
    private int productMenuListId;
    @CreationTimestamp
    @Column(name = "selectionTime")
    private String selectionTime;
    @Column(name = "quantity")
    private int quantity;
    @Column(name = "customerOrderId")
    private int customerOrderId;
    @Column(name = "productMenuId")
    private int productMenuId;
}

File: front/src/app/model/odSystem/product.menu.list.ts
export class ProductMenuList {
  productMenuListId: number;
  selectionTime: string;
  quantity: number;
  customerOrderId: number;
  productMenuId: number;

  constructor(productMenuListId: number, selectionTime: string, quantity: number, customerOrderId: number, productMenuId: number) {
    this.productMenuListId = productMenuListId;
    this.selectionTime = selectionTime;
    this.quantity = quantity;
    this.customerOrderId = customerOrderId;
    this.productMenuId = productMenuId;
  }

  static fromJson(jsonProductMenuList: ProductMenuList) {
    return new ProductMenuList(jsonProductMenuList.productMenuListId, jsonProductMenuList.selectionTime,
      jsonProductMenuList.quantity, jsonProductMenuList.customerOrderId, jsonProductMenuList.productMenuId);
  }
}

